polaris/ocean/convergence/spherical/analysis.py

```diff
@@ -0,0 +1,332 @@
+import datetime
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from polaris import Step
+from polaris.ocean.resolution import resolution_to_subdir
+
+
+class SphericalConvergenceAnalysis(Step):
+    """
+    A step for analyzing the output from the geostrophic convergence test
+
+    Attributes
+    ----------
+    resolutions : list of float
+        The resolutions of the meshes that have been run
+
+    icosahedral : bool
+        Whether to use icosahedral, as opposed to less regular, JIGSAW
+        meshes
+
+    dependencies_dict : dict of dict of polaris.Steps
+        The dependencies of this step
+    """
+    def __init__(self, component, resolutions, icosahedral, subdir,
+                 dependencies, convergence_vars):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component the step belongs to
+
+        resolutions : list of float
+            The resolutions of the meshes that have been run
+
+        icosahedral : bool
+            Whether to use icosahedral, as opposed to less regular, JIGSAW
+            meshes
+
+        subdir : str
+            The subdirectory that the step resides in
+
+        dependencies : dict of dict of polaris.Steps
+            The dependencies of this step
+        """
+        super().__init__(component=component, name='analysis', subdir=subdir)
+        self.resolutions = resolutions
+        self.icosahedral = icosahedral
+        self.dependencies_dict = dependencies
+        self.convergence_vars = convergence_vars
+
+        for _, var in convergence_vars.items():
```

Comment:
```suggestion
        for var in convergence_vars.values():
```

Reply:
Although it seems like `convergence_vars` should be a list as I remarked below.  In that case, just:
```suggestion
        for var in convergence_vars:
```